### PR TITLE
Update upgrade.md

### PR DESCRIPTION
**Pull request by lfbrock**

### CODE REVIEW

File: source/administration/upgrade.md [hunk 1]
```diff
@@ -134,19 +133,15 @@ For any issues, Mattermost Enterprise Edition subscribers and trial license user
...
-#### Upgrade Enterprise Edition to 3.1.x
+#### Upgrade Enterprise Edition to 3.1.x and later
 
 1. Download the **appropriate next upgrade** of your Team Edition server and note any compatibility procedures
       1. Run `platform -version` to check the current version of your Mattermost server
       2. Determine the appropriate next upgrade for your server:
-          - Mattermost `v3.0.x` and `v3.1.x` can upgrade directly to Mattermost `v3.2.x`.
+          - Mattermost `v3.0.x` and above can upgrade directly to Mattermost `v3.4.x`.
           - Mattermost `v2.2.x` can upgrade directly to `v3.1` or `v3.2` but must follow the [extended upgrade guide for `v3.0.x`](https://docs.mattermost.com/administration/upgrade.html#upgrade-to-enterprise-edition-3-0-x)   
+            - Note: For `v3.2.x` only, `"BindUsername"`, and `"BindPassword"` are required fields with anonymous bind not supported.
```
Comment (jasonblais) [inline source/administration/upgrade.md]: should we say these are for LDAP settings?

File: source/administration/upgrade.md [hunk 2]
```diff
@@ -134,19 +133,15 @@ For any issues, Mattermost Enterprise Edition subscribers and trial license user
...
+For `v3.2.x` and `v3.3.x` only, `"FirstNameAttribute"` and `"LastNameAttribute"` are required fields.
```
Comment (jasonblais) [inline source/administration/upgrade.md]: same here

**Comment by jasonblais** [comment]
See above +  adding a note that old public link to files will break

**Comment by esethna** [comment]
+1. Tweaked with JB's comments

**Comment by jasonblais** [comment]
+1

**Pull request merged**